plyer/platforms/linux/wifi.py

```diff
@@ -80,15 +108,18 @@ def _connect(self, network, parameters):
             password = parameters['password']
             cell = self.names[network]
             result = wifi.Scheme.for_cell(
-                'wlan0', network, cell, password
+                self.ifname, network, cell, password
             )
         return result
 
     def _disconnect(self):
         '''
         Disconnect all the networks managed by Network manager.
         '''
-        return call(['nmcli', 'nm', 'enable', 'false'])
+        if self.nmcliVersion > 98:
+            return call(['nmcli', 'networking', 'off'])
```

Comment:
I noticed the command is wrong. You are not ***disconnecting*** from the current network, you are disabling the whole networking interface.